docs/getting-started/tutorials/table_api.md

```diff
@@ -0,0 +1,250 @@
+Tables can support batch queries, streaming queries, or both. 
+
+{% highlight java %}
+tEnv.registerTableSource("transactions", new TransactionTableSource());
+tEnv.registerTableSink("stdout", new StdOutTableSink());
+{% endhighlight %}
+
+We register two tables, a `transactions` input table and a `stdout` output table. 
+The `transactions` table lets us read credit card transactions, which contain `accountId`'s, `timestamp`'s, and dollar `amount`'s. 
+The `stdout` table writes the output of a job to standard out so we can easily see our results. 
+Both tables support running batch and streaming applications.
+
+#### Registering A UDF
+
+Along with tables we include a user defined function for working with timestamps. Our function takes a timestamp and rounds it down the nearest hour. 
+
+{% highlight java %}
+tEnv.registerFunction("truncateDateToHour", new TruncateDateToHour());
+{% endhighlight %}
+
+#### The Query
+
+With our environment configured and tables registered we are ready to build our first application.
+From the `TableEnvironment` we can `scan` an input table to read its rows and then write those results into an output table using `insertInto`. 
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .insertInto("stdout");
+{% endhighlight %}
+
+This intial job simply reads all transactions and writes them to standard out. 
+
+#### Execute
+
+Flink applications are built lazily and shipped to the cluster for execution only once fully formed. 
+We call `ExecutionEnvironment#execute` to begin the execution of our job. 
+
+{% highlight java %}
+env.execute("Spend Report");
+{% endhighlight %}
+
+## Attempt One
+
+Now that we have the skeleton of a job, let's add some business logic.
+We want a report that shows the total spend for each account across each hour of the day.
+Just like a SQL query, we can select the required fields and group by our keys. 
+Because the timestamp field has millisecond granularity we will use our UDF to round it down to the nearest hour.
+Finally, we will select all the fields, summing the total spend per account-hour pair.
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .select("accountId, timestamp.truncateDateToHour as timestamp, amount")
+    .groupBy("accountId, timestamp")
+    .select("accountId, timestamp, amount.sum as total")
+    .insertInto("stdout");
+{% endhighlight %}
+
+## Adding Windows
+
+While this works, we can do better.
+The `timestamp` column represents the [event time]({{ site.baseurl }}/dev/event_time.html) of each row, where event time is the logical time when an event took place in the real world.
+Flink understands the concept of event time, which we can leverage to clean up our code. 
+
+Bucketing data based on time is very common operation in data processing, especially when working with infinite streams. 
+A bucket based on time is called a `Window` and Flink offers flexible windowing semantics. 
```

Comment:
Technically, we also have non-time windows in Flink. Maybe simnply change to `TimeWindow`?